airstack-modules/modules/airstack/domain-name/domain-name.ts

```diff
@@ -0,0 +1,1152 @@
+import {
+  BigInt,
+  Bytes,
+  crypto,
+  log,
+  ByteArray,
+  ens,
+} from "@graphprotocol/graph-ts";
+
+import {
+  AirAccount,
+  AirToken,
+  AirBlock,
+  AirDomain,
+  AirDomainOwnerChangedTransaction,
+  AirDomainTransferTransaction,
+  AirResolver,
+  AirDomainNewResolverTransaction,
+  AirDomainNewTTLTransaction,
+  AirNameRegisteredTransaction,
+  AirNameRenewedTransaction,
+  AirAddrChanged,
+  AirPrimaryDomainTransaction,
+  ReverseRegistrar,
+  PrimaryDomain,
+} from "../../../generated/schema";
+import { uint256ToByteArray, AIR_SET_PRIMARY_DOMAIN_ENTITY_COUNTER_ID, AIR_DOMAIN_OWNER_CHANGED_ENTITY_COUNTER_ID, AIR_ADDR_CHANGED_ENTITY_COUNTER_ID, AIR_NAME_RENEWED_ENTITY_COUNTER_ID, AIR_NAME_REGISTERED_ENTITY_COUNTER_ID, AIR_DOMAIN_NEW_TTL_ENTITY_COUNTER_ID, AIR_DOMAIN_NEW_RESOLVER_ENTITY_COUNTER_ID, AIR_DOMAIN_TRANSFER_ENTITY_COUNTER_ID, ROOT_NODE, ZERO_ADDRESS, ETHEREUM_MAINNET_ID } from "./utils";
+import { BIGINT_ONE, BIG_INT_ZERO, EMPTY_STRING, updateAirEntityCounter, getOrCreateAirBlock } from "../common";
+
+export namespace domain {
+  /**
+   * @dev This function tracks a domain owner change transaction
+   * @param blockHeight block number in the chain
+   * @param blockHash block hash
+   * @param blockTimestamp block timestamp
+   * @param logIndex txn log index
+   * @param chainId chain id
+   * @param newOwner specifies the new owner of the domain
+   * @param transactionHash transaction hash
+   * @param isMigrated specifies if the domain is migrated
+   * @param node specifies the node of the domain
+   * @param label specifies the label of the domain
+   * @param tokenAddress contract address of nft token
+   * @param fromOldRegistry specifies if the event is from the old registry
+   */
+  export function trackDomainOwnerChangedTransaction(
+    blockHeight: BigInt,
+    blockHash: string,
+    blockTimestamp: BigInt,
+    logIndex: BigInt,
+    chainId: string,
+    newOwner: string,
+    transactionHash: string,
+    isMigrated: boolean,
+    node: Bytes,
+    label: Bytes,
+    tokenAddress: string,
+    fromOldRegistry: boolean,
+  ): void {
+    let block = getOrCreateAirBlock(chainId, blockHeight, blockHash, blockTimestamp);
+    let domainId = createAirDomainEntityId(node, label);
+    let domain = getOrCreateAirDomain(new Domain(domainId, chainId, block, tokenAddress));
+    if (fromOldRegistry && domain.isMigrated == true) {
```

Comment:
no, the code checks if the domain linked to the event is already migrated to new contract or not. If it is and the event if from old registry, we don't need to process the event.
Otherwise it sets the isMigrated field as true or false depending on if the event is from new or old contract.